Kernel/Net/RTL8168NetworkAdapter.cpp

```diff
@@ -192,6 +192,51 @@ UNMAP_AFTER_INIT RefPtr<RTL8168NetworkAdapter> RTL8168NetworkAdapter::try_to_ini
     return adopt_ref_if_nonnull(new (nothrow) RTL8168NetworkAdapter(address, irq));
 }
 
+bool RTL8168NetworkAdapter::determine_supported_version()
+{
+    switch (m_version) {
+    case ChipVersion::Version1:
+    case ChipVersion::Version2:
+    case ChipVersion::Version3:
+        return true;
+    case ChipVersion::Version4:
+    case ChipVersion::Version5:
+    case ChipVersion::Version6:
+    case ChipVersion::Version7:
+    case ChipVersion::Version8:
+    case ChipVersion::Version9:
+    case ChipVersion::Version10:
+    case ChipVersion::Version11:
+    case ChipVersion::Version12:
+    case ChipVersion::Version13:
+    case ChipVersion::Version14:
+        return false;
+    case ChipVersion::Version15:
+        return true;
+    case ChipVersion::Version16:
+        return false;
+    case ChipVersion::Version17:
+        return true;
+    case ChipVersion::Version18:
+    case ChipVersion::Version19:
+    case ChipVersion::Version20:
+    case ChipVersion::Version21:
+    case ChipVersion::Version22:
+    case ChipVersion::Version23:
+    case ChipVersion::Version24:
+    case ChipVersion::Version25:
+    case ChipVersion::Version26:
+    case ChipVersion::Version27:
+    case ChipVersion::Version28:
+    case ChipVersion::Version29:
+        return false;
+    case ChipVersion::Version30:
+        return true;
+    default:
+        VERIFY_NOT_REACHED();
```

Comment:
I dont think we should crash for an unknown version (ChipVersion::Unknown), we should just return false as well.

Reply:
Right, I didn't take that into consideration.